design-documents/frontend/technical-vision.md

```diff
@@ -1,144 +1,185 @@
+
+_Disclaimer: there may be licensing limitations that prevent an open-source product from implementing a paid-for product._
+
+### Downloading a 3rd-party package
+
+There are times when you may consider downloading a 3rd party package (e.g., a component). When this happens, consider the following criteria before making a final choice:
+
+1. First thing's first – what is the license?
+   - Is it open source?
+   - You may need to reach out to your legal department if things are a bit blurry.
+1. Does the package have a bright future?
+   - Popularity – are there a significant number of weekly downloads?
+   - Is the downloads line graph going up or down?
+1. Look for tests, ideally with 100% test coverage.
+1. Look at the bundle size with [Bundle Phobia](https://bundlephobia.com/).
+1. Does it have good/clear documentation?
+1. Is it currently maintained?
+   - Look at the number of issues vs. pull requests.
+   - How long have these issues and PRs been open?
+   - When was the last comment made on them and by whom?
+   - **Reliability** – how serious are the issues?
+1. Search and compare this package with other similar packages to gauge which package is the best choice and feel free to reach out to other team members for guidance.
+
+### Resources
+- [spectrum-css](http://opensource.adobe.com/spectrum-css/2.6.0/docs/)
+
+## Styles
+
+Bundle optimization works best when CSS Modules are imported directly into the JavaScript components that use them (a better [TypeScript][] experience is being talked about in [this CRA issue](https://github.com/facebook/create-react-app/issues/5677)).
+
+The coupling here is a bit unfortunate for some projects (e.g., libraries), but pretty standard for applications. For ultimate **portability**, reusable components should be provided their styles externally.
+
+### Resources
+
+- [Adding CSS Modules to CRA](https://facebook.github.io/create-react-app/docs/adding-a-css-modules-stylesheet)
+- [css-modules-typescript-loader](https://www.npmjs.com/package/css-modules-typescript-loader)
+- [CSS Modules VSCode extension](https://marketplace.visualstudio.com/items?itemName=clinyong.vscode-css-modules)
+
+## Accessibility
+
+Ideally, **accessibility** will have been thoroughly considered during the Design/UX stage of a project, but this is not always realistic and many people consider it an afterthought.
+
+To prevent having to completely refactor things when **accessibility** becomes a priority, there are some simple steps you can take:
+
+1. Start scaffolding out a component as if it were an HTML 1.0 form with no JavaScript or CSS. You might have to go back in time to think like this, but you end up getting a lot of accessibility for free. For example, the color and size selections on an e-commerce product page can be simply implemented as radio buttons. All of the keyboard and <kbd>tab</kbd> navigation comes for free! Most anything else people come up with is an anti-pattern and is unexpected by the user.
```

Comment:
> Start scaffolding out a component as if it were an HTML 1.0 form

Recommend rephrasing to drop HTML 1.0 reference here, since there were deprecations and changes to forms between 1.0 >> Living Standard. Example: `name` attribute used for form identity in 1.0-4.0, switched to `id` in 5. Maybe instead just provide an example of how HTML has good a11y defaults?

I'd also recommend rephrasing "as if it were an HTML 1.0 form" because not everything is a form. For example, a photo slider should not be wrapped in a form, but it is critical that it is an accessible widget.

Reply:
Resolved [here](https://github.com/magento/architecture/pull/67/commits/18397d204f1e90a43700f2dc332f68a8d66e21a2#diff-13955aa5a2f331269e98555b510dc546R131).